### ISSUE TITLE
Getting the file size

### ISSUE THREAD

**Issue by AndyJBuchanan** [maziac/binary-file-viewer#15]
I'm writing a parser that needs to know the file size in order to workout the sizes of several blocks. it's a daft format, but that's what I have to work with.

The public API does not seem to present this information, though it looks easy enough to expose.

For now this is my clunky workaround:

   function determineFileSize(){
        const savedOffset = getOffset();
        setOffset(0);
        read();
        read(0);
        let fileSize = getOffset();
        setOffset(savedOffset);
        return fileSize;
    }

FYI. The requirement to add in a read(0) held me up me for a bit

Am I missing a more obvious/nicer solution? If not, would you consider exposing the file size in the API?

Cheers, Andy

**Comment by maziac** [comment]
In registerFileType you get the fileData argument.
You can use it to extract the file size (see the help):
~~~js
 registerFileType((fileExt, filePath, fileData) => {
  ...
  var filesize = fileData.getFileSize();
});
~~~

**Comment by AndyJBuchanan** [comment]
Indeed, but I don't see any way to pass that to the parser function?

**Comment by maziac** [comment]
What if you just declare filesize in global scope:
~~~js
var filesize;
registerFileType((fileExt, filePath, fileData) => {
  ...
  filesize = fileData.getFileSize();
});
~~~

Is it not working?

**Comment by AndyJBuchanan** [comment]
No, that doesn't work. ( Just double checked! )

**Comment by maziac** [comment]
Oh, I See. My mistake. The registerFileType is run in a different context.
However, I think your solution is fine. You could, of course, simplify it a little bit if the first thing that you do in 'registerParser' is reading the file size:
~~~js
registerParser(() => {
	read();
	read(0);
	let filesize = getOffset();
	setOffset(0);
        ...
~~~

And registerParser can work with global variables. I.e. this would work as well:
~~~js
let filesize;
registerParser(() => {
	read();
	read(0);
	filesize = getOffset();
	setOffset(0);
        ...
~~~

**Comment by maziac** [comment]
I'm thinking of introducing a function that returns the remaining data size.
I.e. at the start of parsing you could read it to get the original file size:
~~~js
registerParser(() => {
	let filesize = getRemainingSize();  // file size
        read(100);
        let remaining = getRemainingSize();  // file size - 100
        ...
~~~

Would that help?

**Comment by AndyJBuchanan** [comment]
Yes, that would be quite helpful. Thanks!

**Comment by maziac** [comment]
Implemented in 1.9.0 as getRemainingSize().
